src/View/AbstractDirectoryView.vala

```diff
@@ -312,7 +312,17 @@ namespace FM {
                     view.queue_draw ();
                 }
             });
+
+            in_trash = slot.directory.is_trash;
+            in_recent = slot.directory.is_recent;
+
             model = GLib.Object.@new (FM.ListModel.get_type (), null) as FM.ListModel;
+            if (in_recent)
+                model.set_sort_column_id (get_column_id_from_string ("modified"), Gtk.SortType.DESCENDING);
+            else if (slot.directory.file.info != null) {
+                model.set_sort_column_id (slot.directory.file.sort_column_id, slot.directory.file.sort_order);
+            }
```

Comment:
It's a while since I wrote this but ... I think it is related to the second paragraph of the PR description (unexpected sort order change on clearing)